sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -24,176 +23,253 @@ OTTF_DEFINE_TEST_CONFIG();
+                           /*mpec_upper=*/(uint32_t)kSingleFaultAddrUpper,
+                           /*last_data_addr=*/(uint32_t)&addr_val,
+                           /*curr_pc=*/(uint32_t)kSingleFaultAddrCurrentPc,
+                           /*next_pc=*/(uint32_t)kSingleFaultAddrNextPc);
+
+      LOG_INFO("Setting up watch dog and triggering a double fault.");
+      uint32_t bark_cycles = aon_timer_testutils_get_aon_cycles_from_us(100);
+      uint32_t bite_cycles = aon_timer_testutils_get_aon_cycles_from_us(100);
+
+      // Set wdog as a reset source.
+      CHECK_DIF_OK(dif_pwrmgr_set_request_sources(
+          &pwrmgr, kDifPwrmgrReqTypeReset, kDifPwrmgrResetRequestSourceTwo,
+          kDifToggleEnabled));
+      // Setup the wdog bark and bite timeouts.
+      aon_timer_testutils_watchdog_config(&aon_timer, bark_cycles, bite_cycles,
+                                          false);
+      // Enable cpu info
+      CHECK_DIF_OK(dif_rstmgr_cpu_info_set_enabled(&rstmgr, kDifToggleEnabled));
+
+      double_fault = true;
+      OT_ADDRESSABLE_LABEL(kDoubleFaultFirstAddrLower);
+      addr_val = *kIllegalAddr1;
+      OT_ADDRESSABLE_LABEL(kDoubleFaultFirstAddrUpper);
+      LOG_ERROR(
+          "This should be unreachable; a double fault should have occured.");
+      break;
+
+    case kDifRstmgrResetInfoWatchdog:  // The power-up after the double fault.
+      LOG_INFO("Checking CPU info dump after double fault.");
+
+      dump = get_dump(&ibex);
+
+      CHECK(dump.double_fault == kDifToggleEnabled,
+            "CPU Info dump doesn't show a double fault has happened.");
+
+      // The current behaviour after a double fault is to capture, in the CPU
+      // info dump, the interrupt vector below the one which was taken to jump
+      // to the exception handler as the current PC and the start of the
+      // exception handler as the next PC. This feels wrong. However, with a
+      // lack of a clear definition of what these values should contain, the
+      // test enforces this behaviour so that regressions can be caught. This
+      // behaviour will be double checked at a later date.
+      uint32_t curr_pc = (uint32_t)&_ottf_interrupt_vector + 4;
+      uint32_t next_pc = (uint32_t)&handler_exception;
+
+      check_current_values(&dump,
+                           /*last_exception_addr=*/(uint32_t)kIllegalAddr2,
+                           /*mpec_lower=*/(uint32_t)kDoubleFaultSecondAddrLower,
+                           /*mpec_upper=*/(uint32_t)kDoubleFaultSecondAddrUpper,
+                           /*last_data_addr=*/(uint32_t)kIllegalAddr2,
+                           /*curr_pc=*/curr_pc,
+                           /*next_pc=*/next_pc);
+      check_previous_values(
+          &dump,
+          /*last_exception_addr=*/(uint32_t)kIllegalAddr1,
+          /*mpec_lower=*/(uint32_t)kDoubleFaultFirstAddrLower,
+          /*mpec_upper=*/(uint32_t)kDoubleFaultFirstAddrUpper);
+      break;
```

Comment:
cannot comment on an already existing line but please consider `return`ing `false` by default at the end of the function and `return`in `true` here so that any new branches fail by default in the future. I'm not sure how critical it is to disable the timer though. I don't think our tests are expected to clean-up after themselves.

Reply:
That is a good point the default is not `return false`. I have left in the clean-up for the success case just in-case.